src/psyclone/psyir/nodes/dynamic_omp_task_directive.py

```diff
@@ -0,0 +1,1998 @@
+                             region.
+        :type private_list: List of :py:class:`psyclone.psyir.nodes.Reference`
+        :param firstprivate_list: The list of firstprivate References used in
+                                  this task region.
+        :type firstprivate_list: List of
+                                 :py:class:`psyclone.psyir.nodes.Reference`
+        :param shared_list: The list of shared References for this task.
+        :type shared_list: List of :py:class:`psyclone.psyir.nodes.Reference`
+        :param in_list: The list of input References for this task.
+        :type in_list: List of :py:class:`psyclone.psyir.nodes.Reference`
+        :param out_list: The list of output References for this task.
+        :type out_list: List of :py:class:`psyclone.psyir.nodes.Reference`
+        """
+        lhs = node.children[0]
+        rhs = node.children[1]
+        # Evaluate LHS
+        self._evaluate_write_reference(
+            lhs, private_list, firstprivate_list, shared_list, out_list
+        )
+
+        # Evaluate RHS
+        references = rhs.walk(Reference)
+
+        # If RHS involves a parent loop variable, then our lhs node is a proxy
+        # loop variable.
+        # This handles the case where we have a parent loop, e.g.
+        # do i = x, y, 32
+        # and we have code inside the task region which does
+        # my_temp_var = i+1
+        # array(my_temp_var) = ...
+        # In this case, we need to have this variable as an extra copy of
+        # the proxy parent loop variable. Additionally, there can be multiple
+        # value set for this variable, so we need to store all possible ones
+        # (as they could occur inside an if/else statement and both values)
+        # be visible for each set.
+        added = False
+        for ref in references:
+            if isinstance(ref.parent, ArrayMixin):
+                continue
+            for index, parent_var in enumerate(self._parent_loop_vars):
+                if ref.symbol == parent_var:
+                    if lhs.symbol in self._proxy_loop_vars:
+                        if (
+                            rhs
+                            not in self._proxy_loop_vars[lhs.symbol][
+                                "parent_node"
+                            ]
+                        ):
+                            self._proxy_loop_vars[lhs.symbol][
+                                "parent_node"
+                            ].append(rhs.copy())
+                    else:
+                        subdict = {}
+                        subdict["parent_var"] = parent_var
+                        subdict["parent_node"] = [rhs.copy()]
+                        subdict["loop"] = node
+                        subdict["parent_loop"] = self._parent_loops[index]
+
+                        self._proxy_loop_vars[lhs.symbol] = subdict
+                    added = True
+            if added:
+                break
```

Comment:
So we break out as soon as we find one reference on the RHS that is a parent_var? Pleaes add a comment.

Reply:
Yes - added.

Reply:
Please add the same comment at L1455.

Reply:
Done